sway/input/cursor.c

```diff
@@ -194,15 +366,27 @@ void cursor_send_pointer_motion(struct sway_cursor *cursor, uint32_t time_msec,
 		}
 	}
 
-	// reset cursor if switching between clients
-	struct wl_client *client = NULL;
-	if (surface != NULL) {
-		client = wl_resource_get_client(surface->resource);
-	}
-	if (client != cursor->image_client) {
+	// Handle cursor image
+	if (surface) {
+		// Reset cursor if switching between clients
+		struct wl_client *client = wl_resource_get_client(surface->resource);
+		if (client != cursor->image_client) {
+			wlr_xcursor_manager_set_cursor_image(cursor->xcursor_manager,
+					"left_ptr", cursor->cursor);
+			cursor->image_client = client;
+		}
+	} else if (c && container_is_floating(c)) {
+		// Try a floating container's resize edge
+		enum wlr_edges edge = find_resize_edge(c, cursor);
+		const char *image = edge == WLR_EDGE_NONE ?
+			"left_ptr" : wlr_xcursor_get_resize_name(edge);
+		wlr_xcursor_manager_set_cursor_image(cursor->xcursor_manager, image,
```

Comment:
This will set the cursor each time the mouse moves. This is very inefficient as this re-uploads the image each time. We could probably save the current cursor image name in `sway_cursor`.